SalesforceCommerceCloud/pwa-kit
# W-16094413 Does not show swatch of unorderable products

# Description
This PR attempted to fix the problem with unvalid swatch showing up on PLP tile. 
## Problem
The price shows infinity sign for a swatch on some products. This is because some certain variant is not orderable and it does not have a price, which causes the pricing logic to not be able to find the associate variant to display price correctly.
![image](https://github.com/user-attachments/assets/5e3133c6-b38a-4841-a7dd-de3b888125cb)

## Solution
We will filter out the swatch of unorderable/unvalid variant and not show them on UI. We do the same for PDP for consistency
![image](https://github.com/user-attachments/assets/a9bb37cf-85bc-4a1a-bf8e-510f7a216217)



# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- (change1)

# How to Test-Drive This PR

- Check out code
- Run Retail app 
- Go to this page http://localhost:3000/global/en-GB/search?q=classic%20wrap
- Observe that the invalid swatch is no long displayed on product tile and PDP

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
